provider/resource_configuration.go

```diff
@@ -210,12 +217,25 @@ func resourceConfigurationCreate(d *schema.ResourceData, meta any) error {
 		}
 	}
 
+	// List of extensions represented as a list of configuration.ResourceConfig's
+	// with only the name field set.
```

Comment:
The call to `d.Get` must check if the result is nil. If it is not nil, it is guaranteed that the underlying types are strings.